### DISCUSSION TITLE
Apache log4j Vulnerability CVE-2021-4428

### DISCUSSION THREAD

**Discussion by greenandroid** [dbeaver/dbeaver#14878] [Q&A] [answered]
I see that /product/debug/DBeaver.product.launch has reference to log4j. Thoughts on impact to DBeaver due to this vulnerability?

**Comment by HunterBlacker** [comment, marked as answer]
Since DBeaver CE is an Eclipse-based platform, you can see all of the Java Packages that are loaded by the application or its plugins. Based on DBeaver CE v21.3.2, I do not have any references to log4j. Rather, I note that it is using Apache Commons Logging (org.apache.commons.logging). 

If you want to check your install, go to Help > Installation Information and click on the Configuration tab. Give it a few seconds to load and then use the search bar to find any references to log4j.  Please share if anyone finds a reference.

**Log4J Not Found**

> *** Date: Monday, December 13, 2021 at 9:50:01 AM Eastern Standard Time
*** Platform Details:
*** System properties:
*** System environment variables:
*** Features:
*** Plug-in Registry:
*** User Preferences:
*** Current Install Configuration:

**Commons Logging Found**

> *** Date: Monday, December 13, 2021 at 9:50:01 AM Eastern Standard Time 
*** Platform Details: 
*** System properties: 
org.osgi.framework.system.packages=com.sun.management, com.sun.net.httpserver, com.sun.net.httpserver.spi, com.sun.nio.file, com.sun.security.auth, com.sun.security.auth.callback, com.sun.security.auth.login, com.sun.security.auth.module, com.sun.security.jgss, com.sun.tools.attach, com.sun.tools.attach.spi, java.applet, java.awt, java.awt.color, java.awt.datatransfer, java.awt.desktop, java.awt.dnd, java.awt.event, java.awt.font, java.awt.geom, java.awt.im, java.awt.im.spi, java.awt.image, java.awt.image.renderable, java.awt.print, java.beans, java.beans.beancontext, java.io, java.lang, java.lang.annotation, java.lang.invoke, java.lang.management, java.lang.module, java.lang.ref, java.lang.reflect, java.math, java.net, java.net.spi, java.nio, java.nio.channels, java.nio.channels.spi, java.nio.charset, java.nio.charset.spi, java.nio.file, java.nio.file.attribute, java.nio.file.spi, java.rmi, java.rmi.activation, java.rmi.dgc, java.rmi.registry, java.rmi.server, java.security, java.security.acl, java.security.cert, java.security.interfaces, java.security.spec, java.sql, java.text, java.text.spi, java.time, java.time.chrono, java.time.format, java.time.temporal, java.time.zone, java.util, java.util.concurrent, java.util.concurrent.atomic, java.util.concurrent.locks, java.util.function, java.util.jar, java.util.logging, java.util.prefs, java.util.regex, java.util.spi, java.util.stream, java.util.zip, javax.accessibility, javax.crypto, javax.crypto.interfaces, javax.crypto.spec, javax.imageio, javax.imageio.event, javax.imageio.metadata, javax.imageio.plugins.bmp, javax.imageio.plugins.jpeg, javax.imageio.plugins.tiff, javax.imageio.spi, javax.imageio.stream, javax.management, javax.management.loading, javax.management.modelmbean, javax.management.monitor, javax.management.openmbean, javax.management.relation, javax.management.remote, javax.management.remote.rmi, javax.management.timer, javax.naming, javax.naming.directory, javax.naming.event, javax.naming.ldap, javax.naming.spi, javax.net, javax.net.ssl, javax.print, javax.print.attribute, javax.print.attribute.standard, javax.print.event, javax.rmi.ssl, javax.script, javax.security.auth, javax.security.auth.callback, javax.security.auth.kerberos, javax.security.auth.login, javax.security.auth.spi, javax.security.auth.x500, javax.security.cert, javax.security.sasl, javax.sound.midi, javax.sound.midi.spi, javax.sound.sampled, javax.sound.sampled.spi, javax.sql, javax.sql.rowset, javax.sql.rowset.serial, javax.sql.rowset.spi, javax.swing, javax.swing.border, javax.swing.colorchooser, javax.swing.event, javax.swing.filechooser, javax.swing.plaf, javax.swing.plaf.basic, javax.swing.plaf.metal, javax.swing.plaf.multi, javax.swing.plaf.nimbus, javax.swing.plaf.synth, javax.swing.table, javax.swing.text, javax.swing.text.html, javax.swing.text.html.parser, javax.swing.text.rtf, javax.swing.tree, javax.swing.undo, javax.transaction.xa, javax.xml, javax.xml.catalog, javax.xml.datatype, javax.xml.namespace, javax.xml.parsers, javax.xml.stream, javax.xml.stream.events, javax.xml.stream.util, javax.xml.transform, javax.xml.transform.dom, javax.xml.transform.sax, javax.xml.transform.stax, javax.xml.transform.stream, javax.xml.validation, javax.xml.xpath, jdk.net, jdk.nio, jdk.swing.interop, org.ietf.jgss, org.w3c.dom, org.w3c.dom.bootstrap, org.w3c.dom.css, org.w3c.dom.events, org.w3c.dom.html, org.w3c.dom.ls, org.w3c.dom.ranges, org.w3c.dom.stylesheets, org.w3c.dom.traversal, org.w3c.dom.views, org.w3c.dom.xpath, org.xml.sax, org.xml.sax.ext, org.xml.sax.helpers, sun.misc, sun.reflect 
*** System environment variables: 
*** Features: 
*** Plug-in Registry: 
org.apache.commons.logging (1.2.0.v20180409-1502) "Apache Commons Logging" [Resolved] 
*** User Preferences: 
*** Current Install Configuration: 
Id: org.apache.commons.logging, Version: 1.2.0.v20180409-1502, Location: reference:file:plugins/org.apache.commons.logging_1.2.0.v20180409-1502.jar

**Reply by greenandroid** [reply]
@HunterBlacker Thanks for the confirmation and guidance on this!

**Reply by stgaup** [reply]
As I understand it, the apache.commons.logging is just like an interface, whereas log4j would be one implementation of that interface. This thread states that apache.commons.logging will use log4j if present and configured: 
https://stackoverflow.com/questions/8488762/relation-between-log4j-and-apache-commons-logging/38292596 

Also, this thread from the DBeaver Forum suggests that log4j may be used by drivers that DBeaver relys upon: 
https://dbeaver.io/forum/viewtopic.php?f=2&t=771

So can you really be sure that log4j is not used?

**Reply by p91paul** [reply]
If you install a driver explicitly requiring log4j2-core, then sure you're affected. That's quite unusual though, as a sane driver would depend on an api (slf4j or log4j2-api) not on a specific implementation.

**Comment by LonwoLonwo** [comment]
https://dbeaver.io/2021/12/15/log4shell-vulnerability-is-not-dangerous-for-dbeaver-users/
